HelpSource/Classes/UGen.schelp

```diff
@@ -386,6 +403,62 @@ Complex math support. Returns code::Complex(this, 0.0)::.
 method:: dumpArgs
 Posts a list of the arguments for this UGen and their values.
 
+subsection:: Methods for UGen authors
+
+This is also discussed in  link::Guides/WritingUGens::.
+
+link::Classes/SynthDef:: provides many optimisations that make synths faster and reduce memory consumption.
+There are three main methods to be aware of.
+
+method:: hasObservableEffect
+
+Returns code::true:: or code::false::.
+
+Generally, if you would describe the UGen as 'setting' or 'writing' to something, this should be true.
+
+If a UGen does not indicate that it has an observable effect and is not connected to another UGen which does, it may be removed from the synthdef.
+This is called 'dead code elimination'.
+
+Consider the following example, where line 3 has no effect and can be removed.
+This is only possible because the 'add' UGen (link::Classes/BinaryOpUGen::) sets this method to code::true::
+code::
+SynthDef(\test, {
+    var sin = SinOsc.ar(220);
+    sin + SinOsc.ar(321); // line 3
+    Out.ar(0, sin);
+}).dumpUGens;
+/*
+[0_SinOsc, audio, [220, 0.0], [], [a BinaryOpUGen, an Out], [], []]
+[1_Out, audio, [0, 0_SinOsc], [a SinOsc], [], [], []]
+*/
+::
+
+method:: canBeReplacedByIdenticalCall
```

Comment:
This should be renamed to `canBeReplacedByIdenticalUGen`, no? It is not a function call.